From 7421fb0ddb5e4c540bb740fb672e2823f19d6e10 Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Sat, 21 Feb 2015 14:45:00 -0500
Subject: [PATCH 132/315] display: determine whether or not initial-setup
 should run earlier

We consult whether or not to run initial-setup before we figure it
out.
---
 daemon/gdm-display.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/daemon/gdm-display.c b/daemon/gdm-display.c
index 10c7cbc6..1d20284f 100644
--- a/daemon/gdm-display.c
+++ b/daemon/gdm-display.c
@@ -124,6 +124,7 @@ static void     gdm_display_finalize    (GObject         *object);
 static void     queue_finish            (GdmDisplay      *self);
 static void     _gdm_display_set_status (GdmDisplay *self,
                                          int         status);
+static gboolean wants_initial_setup (GdmDisplay *self);
 G_DEFINE_ABSTRACT_TYPE (GdmDisplay, gdm_display, G_TYPE_OBJECT)
 
 static gboolean
@@ -679,6 +680,8 @@ gdm_display_prepare (GdmDisplay *self)
          */
         look_for_existing_users_sync (self);
 
+        self->priv->doing_initial_setup = wants_initial_setup (self);
+
         g_object_ref (self);
         ret = GDM_DISPLAY_GET_CLASS (self)->prepare (self);
         g_object_unref (self);
@@ -1531,8 +1534,6 @@ gdm_display_set_up_greeter_session (GdmDisplay  *self,
 {
         g_return_if_fail (g_strcmp0 (self->priv->session_class, "greeter") == 0);
 
-        self->priv->doing_initial_setup = wants_initial_setup (self);
-
         if (self->priv->doing_initial_setup) {
                 *username = g_strdup (INITIAL_SETUP_USERNAME);
         } else {
-- 
2.15.1

